### PR TITLE
adds simnode for testing parachain runtime upgrades.

### PR DESCRIPTION
**Pull request by seunlanlege**

### CODE REVIEW

File: integration-tests/simnode/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		let code = picasso_runtime::WASM_BINARY
+			.ok_or("Polkadot development wasm not available")?
+			.to_vec();
+
+		let call = sudo::Call::sudo_unchecked_weight {
+			call: Box::new(frame_system::Call::set_code { code }.into()),
+			weight: 0,
+		};
+		// let su = AccountId32::from_str("5z93WG1Lz47b8AjtVbMaLC4M8rohecXPMSjRDBaMUAbmeCi7")?;
+		let su = AccountId32::from_str("5uAfQTqudXnnSgSMPVowwRjgNFxBDW2d5AQXP2vHDHy2yJ4w")?;
```
Comment (dzmitry-lahoda) [inline integration-tests/simnode/src/main.rs:43]: please reuse key

Comment (dzmitry-lahoda) [reply integration-tests/simnode/src/main.rs:43]: variable

Comment (dzmitry-lahoda) [reply integration-tests/simnode/src/main.rs:43]: of `5uAfQTqudXnnSgSMPVowwRjgNFxBDW2d5AQXP2vHDHy2yJ4w` parsed only once

File: integration-tests/simnode/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		let new_runtime_version = node
+			.client()
+			.executor()
+			.runtime_version(&BlockId::Hash(node.client().info().best_hash))?
+			.spec_version;
+		println!("\n\nnew_runtime_version: {}\n\n", new_runtime_version);
+
+		// just confirming
+		assert!(
+			new_runtime_version > old_runtime_version,
```
Comment (dzmitry-lahoda) [inline integration-tests/simnode/src/main.rs:73]: and new version has version as it is defined in code?

File: runtime/picasso/src/lib.rs [hunk 3]
```diff
@@ -784,7 +784,7 @@ impl Contains<Call> for BaseCallFilter {
 	fn contains(call: &Call) -> bool {
 		#[cfg(feature = "develop")]
 		if call_filter::Pallet::<Runtime>::contains(call) {
-			return false
+			return false;
```
Comment (dzmitry-lahoda) [inline runtime/picasso/src/lib.rs:787]: is it draft pr?

File: integration-tests/simnode/src/main.rs (hunk 2 above)
Comment (seunlanlege) [reply integration-tests/simnode/src/main.rs:73]: as defined in spec version

File: integration-tests/simnode/src/main.rs (hunk 1 above)
Comment (seunlanlege) [reply integration-tests/simnode/src/main.rs:43]: yeah missed that

File: integration-tests/simnode/src/chain_info.rs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		(
+			system::CheckSpecVersion::<Self::Runtime>::new(),
+			system::CheckTxVersion::<Self::Runtime>::new(),
+			system::CheckGenesis::<Self::Runtime>::new(),
+			system::CheckMortality::<Self::Runtime>::from(Era::Immortal),
+			system::CheckNonce::<Self::Runtime>::from(
+				system::Pallet::<Self::Runtime>::account_nonce(from),
+			),
+			system::CheckWeight::<Self::Runtime>::new(),
+			transaction_payment::ChargeTransactionPayment::<Self::Runtime>::from(0),
```
Comment (KaiserKarel) [inline integration-tests/simnode/src/chain_info.rs:69]: This is missing the signed-extra needed for the crowdloan

File: runtime/dali/src/weights/mod.rs [hunk 5]
```diff
@@ -3,21 +3,18 @@ pub mod balances;
 pub mod collator_selection;
 pub mod collective;
 pub mod crowdloan_bonus;
```
Comment (KaiserKarel) [inline runtime/dali/src/weights/mod.rs:5]: Can remove this now?

File: runtime/picasso/src/lib.rs [hunk 6]
```diff
@@ -91,7 +91,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	//   `spec_version`, and `authoring_version` are the same between Wasm and native.
 	// This value is set to 100 to notify Polkadot-JS App (https://polkadot.js.org/apps) to use
 	//   the compatible custom types.
-	spec_version: 200,
+	spec_version: 2000,
```
Comment (KaiserKarel) [inline runtime/picasso/src/lib.rs:94]: This comment is now a bit out of date right?

File: integration-tests/simnode/src/chain_info.rs (hunk 4 above)
Comment (seunlanlege) [reply integration-tests/simnode/src/chain_info.rs:69]: That's in the Dali runtime, it's un audited and hence not on Picasso.

File: runtime/dali/src/weights/mod.rs (hunk 5 above)
Comment (seunlanlege) [reply runtime/dali/src/weights/mod.rs:5]: Ah yes, will do

**Review by KaiserKarel — APPROVED** [review]
LGTM, but please remove the wasm artifact, no need to commit that, although it does not affect the releaser.

**Review by haroldsphinx — APPROVED** [review]

**Review by haroldsphinx — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

**Pull request merged**